pkg/controller/dataimportcron-controller.go

```diff
@@ -1172,6 +1249,94 @@ func InitPollerPodSpec(c client.Client, cron *cdiv1.DataImportCron, podSpec *cor
 		)
 	}
 
+	err = createPodSpec(container, insecureTLS, *cdiConfig, c, podSpec, volumes)
+
+	return err
+}
+
+func initPollerPodSpecHTTP(c client.Client, cron *cdiv1.DataImportCron, podSpec *corev1.PodSpec, image string, pullPolicy corev1.PullPolicy, log logr.Logger) error {
```

Comment:
No idea what you tried to do here but cloning the poller won't work.

Reply:
What exactly won't work? I haven't added complex logic to the poller yet, I just adjusted the code for each type of import.
When using Registry I need `getCronRegistrySource` and with HTTP I need `getCronHTTPSource`, in the Registry the URL is of type `*string` and in HTTP it is of type `string`. So for now I kept the code that was the same and moved the different code to another function. This will probably change later.
If I leave the original code I get an error in the unit test: `Cron with no registry source test-cron.`
I tried going through it again but couldn't find anything else at the moment so I'll try again when I get back, maybe with more information/reviews

Reply:
All you do here is using the existing poller pod, which definately won't work.